web/src/lib/data-hooks/useUserData.ts

```diff
@@ -49,7 +49,7 @@ export const useUserData = (): UseUserDataResponse => {
       setUpdateQueue(new UpdateQueueFactory(data, update));
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  }, [data, updateQueue]);
```

Comment:
This is the needed change - to allow this to run and set the update queue when `userData` exists.

That way, our useEffect runs correctly without having to enter that `else if` statement